### ISSUE TITLE
EPS axis 

### ISSUE THREAD

**Issue by ghalym** [pcdshub/lcls-twincat-motion#137]
Having EPS axis variables added to the DUT.
bEPSAxis, bEPSFwdEnable, bEPSBwdEnable, for collision prevention.

**Comment by ZLLentz** [comment]
Implementation plan:
- extend DUT_MotionStage as suggested
- add logic to FB_SetEnables as appropriate (add `AND (bEPSFwdEnable OR NOT bEPSAxis)` for example)

**Comment by ZLLentz** [comment]
Justification from my offline chat with Maggie:
- EPS is something we do a lot
- We want a difference in the IOC between "limit switch hit" and "EPS restriction"

Note that the limit indicator currently on the screen is a summary of all limits, so we'd need some auxiliary screens to disambiguate the various limits (probably useful in general!)

**Comment by ghalym** [comment]
Any updates on this issue? The reason I am asking is that in RIX and qRIX we have some special EPS consideration and when we link the EPS logic to the bForwardEnable, it gets a bit confusing on the screen when it shows that both limit switches are "Hit" while it's actually the enable signal. I think everyone got used to this to be the only the limit switch indicator.
(@jyotiphy because we were talking about this the other day)

**Comment by ZLLentz** [comment]
I have not done any work on this nor is it on my immediate radar. We can of course rearrange priorities if this is becoming important.

**Comment by ghalym** [comment]
So, my understanding is that QRIX commissioning is in November. So QRIXS will need it some time before then.
@jyotiphy When do you think this feature will be needed for RIX?

**Comment by jyotiphy** [comment]
Yes, it is very confusing by seeing both limit switch hit on the motor screen. chemRIX has three such valves where we are using this feature already. @ZLLentz I’m not sure how big effort is this from your side. Maybe we can live with this for sometime as it’s not breaking anything. But if it’s easy to implement, it might be better to implement this feature sooner than later.

**Comment by ZLLentz** [comment]
We can do it, we need to think a moment about the best way to implement it such that we see the result in the screen without messing up all the other motors and that we hit all the needs/edge cases that we care about.

What are our requirements? Something like:
- Add standard variables for directional interlock enables (fwd, bwd, power) and integrate those into the main enable loop
- Add a string field for describing the interlock
- Connect the interlock variables to epics PVs
- Edit the status so that the low and high limit switch statuses only include the physical switch
- Create alternate classes/screens for interlocked motors (possibly include one for gantry-interlocked motors as well?)
- Make sure existing motors without interlocks can still move as-deployed
- Make sure existing motor screens can still be used as-deployed

**Comment by ZLLentz** [comment]
For the UI, ultimately, we need to identify the best ways to communicate the interlock. It's not enough I think to just make the light not light up, we need a different light to light up.

**Comment by jyotiphy** [comment]
> For the UI, ultimately, we need to identify the best ways to communicate the interlock. It's not enough I think to just make the light not light up, we need a different light to light up.

yes! "different light to light up"

**Comment by ZLLentz** [comment]
Created a jira ticket at https://jira.slac.stanford.edu/browse/LCLSPC-373 to track the overall work since it may need to span multiple repositories
